lib/xgu/xgu.c

```diff
@@ -0,0 +1,29 @@
+#include "xgu.h"
+
+#include <assert.h>
+#include <pbkit/pbkit.h>
+
+DWORD *pushbuffer_edge;
+
+static DWORD* push_command(DWORD command, int nparams) {
+    pb_push(pushbuffer_edge++, command, nparams);
+    return pushbuffer_edge;
+}
+
+static DWORD* push_parameter(DWORD parameter) {
+    *pushbuffer_edge++ = parameter;
+}
+
+DWORD* xgu_set_object(DWORD* p, int instance) {
```

Comment:
Mark these inline - all GPU commands will likely run through these, so this is perf critical, also helps to search & replace to add further attributes later.